### DISCUSSION TITLE
Disconnect / reconnect via intent without confirmation dialog

### DISCUSSION THREAD

**Discussion by DKAC** [strongswan/strongswan#813] [General]
Hi all,

I am looking for a way to silently disconnect via intent from another app without the user having to confirm this. I found the disconnect method here

\src\frontends\android\app\src\main\java\org\strongswan\android\ui\VpnProfileControlActivity.java

where in one line it says /* allow explicit termination without confirmation */. Sadly, I don't really understand on which conditions the disconnect will be performed without confirmation. I would be greatful for any explanation.

Regards, Daniel

**Comment by tobiasbrunner** [comment]
It's based on the UUID of the profile that's to be disconnected. See the "How-to use Intents to..." section on [AndroidVPNClient](https://wiki.strongswan.org/projects/strongswan/wiki/AndroidVPNClient) for details.

**Reply by DKAC** [reply]
Found it (that piece of information was well hidden ^^)... thank you very much!

Btw. I need to sleep for some milliseconds between disconnect and reconnect, otherwise, the dialog still pops up, but that's totally fine with me... just in case someone else is trying to do the same thing.
